### ISSUE TITLE
Support readable aliases for HHClientLinter rules

### ISSUE THREAD

**Issue by fredemmott** [hhvm/hhast#410]
#409

Perhaps we should support `DontAwaitInALoop` as an alias for #409:

- avoids the BC break
- more readable

This would be a divergence from HHAST.

This could also be more general, not just for HHClientLinter - then we could support the same suppression codes within FB with Aurora/HackAST, and externally with HHAST, when the linters do not have the same suppression code at present.

**Comment by Atry** [comment]
I think we don't want to do this manually because the aliases would be out of sync quickly when the upstream code changes. I think the best way to solve this issue is to export the metadata of Rust lint rules to Hack automatically.

**Comment by fredemmott** [comment]
I think we need a manual aliasing method regardless, as the names in ocaml do not necessarily match the historical hhast linter names - requiring renames removes the BC b enefits.

**Comment by Atry** [comment]
I agree we should allow some overridden aliases configured in HHAST code base for historical hhast linter names, but shall we configure a default name in the OCaml code base then generate the corresponding Hack definition, so that we could have readable alias for new OCaml linters by default?

**Comment by fredemmott** [comment]
![image](https://user-images.githubusercontent.com/360927/144097675-0a95d92a-36d7-49c0-b914-d309ce6eb3f8.gif)

Though, if exported from ocaml, should probably default to the ocaml class name rather than needing explicit configuration
